src/rpcblockchain.cpp

```diff
@@ -392,11 +394,36 @@ UniValue getblock(const UniValue& params, bool fHelp)
             "\nExamples:\n"
             + HelpExampleCli("getblock", "\"00000000c937983704a73af28acdec37b049d214adbda81d7e2a3dd146f6ed09\"")
             + HelpExampleRpc("getblock", "\"00000000c937983704a73af28acdec37b049d214adbda81d7e2a3dd146f6ed09\"")
+            + HelpExampleCli("getblock", "12800")
+            + HelpExampleRpc("getblock", "12800")
         );
 
     LOCK(cs_main);
 
     std::string strHash = params[0].get_str();
+
+    // If height is supplied, find the hash
+    if (strHash.size() < 64) {
```

Comment:
Non-blocking: is there a constant somewhere we can reference here instead of a hard-coded length? Or is there a precedent in the codebase for assuming txid hashes are 64 chars long? It looks like `uint256S()` reads up to 64 characters from the end of the given string, so I guess it's fair enough to specify 64 here given we directly reference `uint256` shortly after.